vars/template.go

```diff
@@ -184,13 +184,13 @@ func newVarsTracker(vars Variables, expectAllFound, expectAllUsed bool) varsTrac
 // is var name; 2) 'foo:bar', where foo is var source name, and bar is var name;
 // 3) '.:foo', where . means a local var, foo is var name.
 func (t varsTracker) Get(varName string) (interface{}, bool, error) {
-	t.visitedAll[varName] = struct{}{}
-
 	varRef := parseVarName(varName)
 
+	t.visitedAll[identifier(varRef)] = struct{}{}
+
 	val, found, err := t.vars.Get(VariableDefinition{Ref: varRef})
 	if !found || err != nil {
-		t.missing[varRef.Path] = struct{}{}
+		t.missing[varRef.Name] = struct{}{}
```

Comment:
Just checking - is the reason we use `varRef.Name` rather than `identifier(varRef)` so that our error message matches the var they tried to reference? e.g. if they reference `((vs:var1.path.to.field))`, but `var1` doesn't exist in `vs`, we emit an error like:

```
undefined vars: vs:var1.path.to.field
```

rather than

```
undefined vars: vs:var1
```

??

Reply:
yes I think `name` is the raw variable string that has everything. We'd like to show the error as detailed as possible.

In the other case for the unused error, we rely on the returned list of variable definitions from underlining variable type, where field info is not included. So we have to use `source` + `path` as identifier
https://github.com/concourse/concourse/blob/d8818bdbc1840e64253d3de6ec3c0273324618f3/vars/static_vars.go#L12-L20